be/src/formats/parquet/group_reader.cpp

```diff
@@ -286,6 +279,51 @@ Status GroupReader::_read_range(const std::vector<int>& read_columns, const Rang
     return Status::OK();
 }
 
+StatusOr<size_t> GroupReader::_read_range_round_by_round(const Range<uint64_t>& range, Filter* filter,
+                                                         ChunkPtr* chunk) {
+    const std::vector<int>& round_order = _round_by_round_ctx->get_round_order();
+    size_t round_cost = 0;
+    size_t hit_count = 0;
+    for (int col_idx : round_order) {
+        auto& column = _param.read_cols[col_idx];
+        round_cost += _round_by_round_ctx->get_column_cost(col_idx);
+        SlotId slot_id = column.slot_id;
+        RETURN_IF_ERROR(
+                _column_readers[slot_id]->read_range(range, filter, (*chunk)->get_column_by_slot_id(slot_id).get()));
+
+        if (std::find(_dict_column_indices.begin(), _dict_column_indices.end(), col_idx) !=
+            _dict_column_indices.end()) {
+            SCOPED_RAW_TIMER(&_param.stats->expr_filter_ns);
+            SCOPED_RAW_TIMER(&_param.stats->group_dict_filter_ns);
+            for (const auto& sub_field_path : _dict_column_sub_field_paths[col_idx]) {
+                _column_readers[slot_id]->filter_dict_column((*chunk)->get_column_by_slot_id(slot_id), filter,
+                                                             sub_field_path, 0);
+                hit_count = SIMD::count_nonzero(*filter);
```

Comment:
can we get hit count from `filter_dict_column`

Reply:
the predicate evaluate does not return hit count, we need calculate it.